File: code/go/ch17/src/luago/api/lua_vm.go
package api

type LuaVM interface {
	LuaState
	PC() int
	AddPC(n int)
	Fetch() uint32
	GetConst(idx int)
	GetRK(rk int)
	RegisterCount() int
	LoadVararg(n int)
	LoadProto(idx int)
	CloseUpvalues(a int)
}

File: code/java/ch10/src/main/java/com/github/zxh0/luago/api/LuaVM.java
package com.github.zxh0.luago.api;

public interface LuaVM extends LuaState {

    void addPC(int n);
    int fetch();
    void getConst(int idx);
    void getRK(int rk);
    int registerCount();
    void loadVararg(int n);
    void loadProto(int idx);
    void closeUpvalues(int a);

}
